pkg/registry/flowcontrol/ensurer/strategy.go

```diff
@@ -267,36 +269,33 @@ func EnsureConfiguration[ObjectType configurationObjectType](ctx context.Context
 
 		// we always re-create a missing configuration object
 		if _, err = ops.Create(ctx, ops.DeepCopy(bootstrap), metav1.CreateOptions{FieldManager: fieldManager}); err == nil {
-			klog.V(2).InfoS(fmt.Sprintf("Successfully created %T", bootstrap), "type", configurationType, "name", name)
+			logger.V(2).Info(fmt.Sprintf("Successfully created %T", bootstrap))
 			return nil
 		}
 
 		if !apierrors.IsAlreadyExists(err) {
 			return fmt.Errorf("cannot create %T type=%s name=%q error=%w", bootstrap, configurationType, name, err)
 		}
-		klog.V(5).InfoS(fmt.Sprintf("Something created the %T concurrently", bootstrap), "type", configurationType, "name", name)
+		logger.V(5).Info(fmt.Sprintf("Something created the %T concurrently", bootstrap))
 	}
 
-	klog.V(5).InfoS(fmt.Sprintf("The %T already exists, checking whether it is up to date", bootstrap), "type", configurationType, "name", name)
+	logger.V(5).Info(fmt.Sprintf("The %T already exists, checking whether it is up to date", bootstrap))
 	newObject, update, err := strategy.ReviseIfNeeded(ops, current, bootstrap)
 	if err != nil {
 		return fmt.Errorf("failed to determine whether auto-update is required for %T type=%s name=%q error=%w", bootstrap, configurationType, name, err)
 	}
 	if !update {
-		if klogV := klog.V(5); klogV.Enabled() {
-			klogV.InfoS("No update required", "wrapper", bootstrap.GetObjectKind().GroupVersionKind().Kind, "type", configurationType, "name", name,
-				"diff", cmp.Diff(current, bootstrap))
+		if v5 := logger.V(5); v5.Enabled() {
+			v5.Info("No update required", "wrapper", bootstrap.GetObjectKind().GroupVersionKind().Kind, "diff", cmp.Diff(current, bootstrap))
 		}
 		return nil
 	}
 
-	if _, err = ops.Update(ctx, newObject, metav1.UpdateOptions{FieldManager: fieldManager}); err == nil {
-		klog.V(2).Infof("Updated the %T type=%s name=%q diff: %s", bootstrap, configurationType, name, cmp.Diff(current, bootstrap))
+	if updated, err := ops.Update(ctx, newObject, metav1.UpdateOptions{FieldManager: fieldManager}); err == nil {
+		logger.V(2).Info(fmt.Sprintf("Updated the %T", bootstrap), "diff", cmp.Diff(current, updated))
```

Comment:
The code was doing a Diff that shows the motivation for the update: the difference between what was stored and what was desired. That makes sense to me.

Reply:
It showed a diff of status on every run.

Reply:
We could diff the spec or wipe status.

Reply:
Do you mean that this log statement executed even when there was no change? Or that it executed when the only change was in status? Are you sure there was not a change in the annotation named "apf.kubernetes.io/autoupdate-spec"?

If you look earlier to the call on `strategy.ReviseIfNeeded` you will see that it is indeed just comparing the spec and the aforementioned annotation. If you are seeing cases where it unnecesseary updates are being done then this is a problem in the logic and it should be debugged. This can be a separate PR.

Reply:
If it is called, it will include the status as the status is not set in `bootstrap` because `current` will have it set. Is that correct?

Reply:
`strategy.ReviseIfNeeded` specifically compares only the spec and that one attribute.

Or is your complaint that the log message includes the diff of the whole objects (despite much less being relevant)?

Reply:
Yes, it always has status changes in there. It irritates.